provider/internal/device/asa/resource.go

```diff
@@ -305,14 +299,20 @@ func (r *AsaDeviceResource) Update(ctx context.Context, req resource.UpdateReque
 		return
 	}
 
-	tagsInp := tags.New(util.TFStringListToGoStringList(planData.Labels)...)
+	// convert tf tags to go tags
+	tagsGoList, err := util.TFStringSetToGoStringList(ctx, planData.Labels)
```

Comment:
We could also wrap lines 303-307 and 236-240 into a function?

Reply:
done, also fix similar places 👍